app/models/data_resources/news_item.rb

```diff
@@ -45,10 +46,17 @@ def settings
   end
 
   def find_or_create_category
-    return if category_name.blank?
-
-    category_to_add = Category.where(name: category_name).first_or_create
-    categories << category_to_add unless categories.include?(category_to_add)
+    if category_name.present?
+      category_to_add = Category.where(name: category_name).first_or_create
+      categories << category_to_add unless categories.include?(category_to_add)
+    end
+
+    if category_names.present?
+      category_names.each do |cat|
+        category_to_add = Category.where(name: cat[:name]).first_or_create
+        categories << category_to_add unless categories.include?(category_to_add)
```

Comment:
der body is identisch hier oder? nur der `name` is variabel. könnte eine methode sein, die als return das catrgory_to_add hat und dann sowhol für `category_names` als auch für `category_name` benutzt werden?

Reply:
ignoriere ich bewusst erstmal

Reply:
ok